packages/chat_repository_interface/lib/src/extension/uint8list_data_uri.dart

```diff
@@ -0,0 +1,18 @@
+import "dart:convert";
+import "dart:typed_data";
+import "package:mime/mime.dart";
+
+/// Extension that provides a converter function from
+/// Uin8List to a base64Encoded data uri.
+extension ToDataUri on Uint8List {
+  /// This function converts the Uint8List into
+  /// a uri with a data-scheme.
+  String toDataUri() {
+    var mimeType = lookupMimeType("", headerBytes: this);
+    if (mimeType == null) throw Exception();
```

Comment:
Please either throw a specific descriptive exception, or if this is a programmer error, then use a specific Error implementation